test/Users.test.ts

```diff
@@ -0,0 +1,172 @@
+  });
+
+  it('should return six users starting from the third', async function() {
+    const res = await requestUsersQueryWithToken({ limit: 6, offset: 2 }, correctToken);
+
+    const { totalCount, users, pageInfo } = res.body.data.Users;
+    expect(totalCount).to.be.equals(NUMBER_OF_USERS);
+    expect(users).to.be.deep.equals(graphQLFormatUsers.slice(2, 8));
+    expect(pageInfo).to.be.deep.equals({ hasPreviousPage: true, hasNextPage: true });
+  });
+
+  it('should return the last two users skipping eight', async function() {
+    const res = await requestUsersQueryWithToken({ limit: 5, offset: 8 }, correctToken);
+
+    const { totalCount, users, pageInfo } = res.body.data.Users;
+    expect(totalCount).to.be.equals(NUMBER_OF_USERS);
+    expect(users).to.be.deep.equals(graphQLFormatUsers.slice(8));
+    expect(pageInfo).to.be.deep.equals({ hasPreviousPage: true, hasNextPage: false }); 
+  });
+
+  it('should return zero users because it is skipping all of them', async function() {
+    const res = await requestUsersQueryWithToken({ limit: 5, offset: 100 }, correctToken);
+
+    const { totalCount, users, pageInfo } = res.body.data.Users;
+    expect(totalCount).to.be.equals(NUMBER_OF_USERS);
+    expect(users).to.be.empty;
+    expect(pageInfo).to.be.deep.equals({ hasPreviousPage: true, hasNextPage: false });
+  });
+
+  it('should return default five users because limit is zero', async function() {
+    const res = await requestUsersQueryWithToken({ limit: 0, offset: 0 }, correctToken);
+
+    const { totalCount, users, pageInfo } = res.body.data.Users;
+    expect(totalCount).to.be.equals(NUMBER_OF_USERS);
+    expect(users).to.be.deep.equals(graphQLFormatUsers.slice(0, 5));
+    expect(pageInfo).to.be.deep.equals({ hasPreviousPage: false, hasNextPage: true });
+  });
+
+  it('should return all users because limit is higher than total', async function() {
+    const res = await requestUsersQueryWithToken({ limit: 100, offset: 0 }, correctToken);
+
+    const { totalCount, users, pageInfo } = res.body.data.Users;
+    expect(totalCount).to.be.equals(NUMBER_OF_USERS);
+    expect(users).to.be.deep.equals(graphQLFormatUsers);
+    expect(pageInfo).to.be.deep.equals({ hasPreviousPage: false, hasNextPage: false });
+  });
+
+  it('should return an error because limit is negative', async function() {
+    const res = await requestUsersQueryWithToken({ limit: -1, offset: 0 }, correctToken);
+    
+    const { errors } = res.body;
+    expect(errors[0].details).to.be.deep.equals([{ message: ErrorMessages.LIMIT_NEGATIVE }]);
+  });
+
+  it('should return an error because offset is negative', async function() {
+    const res = await requestUsersQueryWithToken({ limit: 5, offset: -1 }, correctToken);
+
+    const { errors } = res.body;
+    expect(errors[0].details).to.be.deep.equals([{ message: ErrorMessages.OFFSET_NEGATIVE }]);
+  });
+  
+  it('limit should be optional and return five users by default', async function() {
```

Comment:
Poderia comstruir essa frase de modo semelhante à anterior: `should return default five users because limit is zero`